Remove extra-config for cert-manager

rstarmer opened the pull request:
Minikube does the right thing (as of 0.28.1 at least) with creating the embedded CA.  The extra-config parameters appear to have been necessary previously and were resolved to use the "right" credentials built by Minikube directly. In fact, passing those parameters appears to break current minikube deployments, making it impossible to create new service accounts and resources that rely on them. (like a tiller service account for a helm deployment of Istio...)

I found this bug that referenced this issue: https://github.com/kubernetes/minikube/issues/1647 which is now closed.

istio-testing commented:
Hi @rstarmer. Thanks for your PR.

I'm waiting for a [istio](https://github.com/orgs/istio/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/istio/people) to skip this step.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

sdake commented:
@rstarmer you will need to fill out the CLA to have this work merged.

Cheers
-steve

rstarmer commented:
Well, my experience (minikube 0.29.0), was that pilot would end up in a
crashloop complaining about a lack of access to the paths defined. I didn't
dig much further, removed the paths, and everything works...

Robert


On Mon, Oct 1, 2018 at 10:26 PM Steven Dake <notifications@github.com>
wrote:

> *@sdake* approved this pull request.
>
> hey nerd, welcome to the party! @andraxylia
> <https://github.com/andraxylia> had put these back in after I removed
> them. IIRC she had a good reason to do so - but I don't recall the reason.
> Happy to merge if she agrees with the change.
>
> Cheers
> -steve
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio.io/pull/2708#pullrequestreview-160570059>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA43ZqoathzaFw1TXeR2Rw_ZHRzZRhOFks5ugvkYgaJpZM4XBEji>
> .
>

rstarmer commented:
Have done, apparently I submitted the patch on my personal github account,
so I signed a personal one, and one for my company too.
Robert


On Mon, Oct 1, 2018 at 10:27 PM Steven Dake <notifications@github.com>
wrote:

> @rstarmer <https://github.com/rstarmer> you will need to fill out the CLA
> to have this work merged.
>
> Cheers
> -steve
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio.io/pull/2708#issuecomment-426153205>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AA43ZhSSP4ieNYF074GHKS-oiCSC6i3Vks5ugvlJgaJpZM4XBEji>
> .
>

andraxylia commented:
My reasoning was we should advise the same config as tested by circleci. Change was in this PR:

https://github.com/istio/istio.io/pull/2082

At the time, I said  "Checked with ayj and we still need it if we use self-signed certificates or k8s CA. It's safer to leave it there."

Maybe we need them in some case and not in other cases?

rstarmer commented:
I wonder if there is additional configuration needed then, because when I
launch minikube (0.29.0) with the extra commands creating a 1.10.0 K8S
system, the environment doesn't completely form properly. For example, you
can create a service account for helm (which is really a DB operation), but
tiller doesn't launch. And in fact, it appears that other control plane
services aren't properly running.

Robert


On Mon, Oct 1, 2018 at 10:54 PM Andra Cismaru <notifications@github.com>
wrote:

> My reasoning was we should advise the same config as tested by circleci.
> Change was in this PR:
>
> #2082 <https://github.com/istio/istio.io/pull/2082>
>
> At the time, I said "Checked with ayj and we still need it if we use
> self-signed certificates or k8s CA. It's safer to leave it there."
>
> Maybe we need them in some case and not in other cases?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio.io/pull/2708#issuecomment-426157299>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AA43Zt_faDYfirwDRav7UAJzBsuSgZkaks5ugv-LgaJpZM4XBEji>
> .
>

adarshaj commented:
For those who are coming from the istio.io 's page, which doesn't yet reflect this change. The latest minikube's path for certs had to be updated from `/var/lib/localkube` to `/var/lib/minikube`.. (which isn't necessary as demonstrated by this PR), however just leaving here the command that worked for me:

```bash
minikube start --memory=8192 --cpus=4 --kubernetes-version=v1.12.1 \
    --extra-config=controller-manager.cluster-signing-cert-file="/var/lib/minikube/certs/ca.crt" \
    --extra-config=controller-manager.cluster-signing-key-file="/var/lib/minikube/certs/ca.key" \
    --vm-driver=kvm2
```

CodeJjang commented:
@adarshaj I don't see `/var/lib/minikube...` either in version v.0.30.0.  
Where is it located now?
